diff --git a/plugins/tasklist/tasklist-widget.c b/plugins/tasklist/tasklist-widget.c
index 426089fd..e2643165 100644
--- a/plugins/tasklist/tasklist-widget.c
+++ b/plugins/tasklist/tasklist-widget.c
@@ -2278,14 +2278,14 @@ xfce_tasklist_child_new (XfceTasklist *tasklist)
   if (!xfce_tasklist_vertical (tasklist))
     {
       /* gtk_box_reorder_child (GTK_BOX (child->box), child->icon, 0); */
-      gtk_label_set_xalign (GTK_LABEL (child->label), 0.0);
+      gtk_label_set_xalign (GTK_LABEL (child->label), 0.5);
       gtk_label_set_yalign (GTK_LABEL (child->label), 0.5);
       gtk_label_set_ellipsize (GTK_LABEL (child->label), tasklist->ellipsize_mode);
     }
   else
     {
       /* gtk_box_reorder_child (GTK_BOX (child->box), child->icon, -1); */
-      gtk_label_set_yalign (GTK_LABEL (child->label), 0.0);
+      gtk_label_set_yalign (GTK_LABEL (child->label), 0.5);
       gtk_label_set_xalign (GTK_LABEL (child->label), 0.5);
       gtk_label_set_angle (GTK_LABEL (child->label), 270);
       /* TODO can we already ellipsize here yet? */
@@ -4375,14 +4375,14 @@ xfce_tasklist_update_orientation (XfceTasklist *tasklist)
       if (horizontal)
         {
           /* gtk_box_reorder_child (GTK_BOX (child->box), child->icon, 0); */
-          gtk_label_set_xalign (GTK_LABEL (child->label), 0.0);
+          gtk_label_set_xalign (GTK_LABEL (child->label), 0.5);
           gtk_label_set_yalign (GTK_LABEL (child->label), 0.5);
           gtk_label_set_angle (GTK_LABEL (child->label), 0);
         }
       else
         {
           /* gtk_box_reorder_child (GTK_BOX (child->box), child->icon, -1); */
-          gtk_label_set_yalign (GTK_LABEL (child->label), 0.0);
+          gtk_label_set_yalign (GTK_LABEL (child->label), 0.5);
           gtk_label_set_xalign (GTK_LABEL (child->label), 0.5);
           gtk_label_set_angle (GTK_LABEL (child->label), 270);
         }
